### PR TITLE
attempt to add test of Chinese error message

### PR DESCRIPTION
**Pull request by mattdowle**
For #4402, follow up to #4523

### CODE REVIEW

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -16910,3 +16910,14 @@ dt = data.table(time = 1:8, v = INT(5,7,6,1,8,4,2,3))
 dt[time == 2L, v := 2L]
 dt[time == 7L, v := 7L]
 test(2140, dt[dt, on=.(time>time, v>v), .N, by=.EACHI], data.table(time=1:8, v=INT(5,2,6,1,8,4,7,3), N=INT(3,5,2,4,0,1,0,0)))
+
+# repeat of test 450 for #4402
+DT = data.table(a=1:3,b=4:6)
+test(2141.1, rbind(DT,list(c=4L,a=7L)), error="Column 1 ['c'] of item 2 is missing in item 1")
+old = Sys.getenv("LANGUAGE")
+Sys.setenv(LANGUAGE="zh_CN")
```
Comment (MichaelChirico) [inline inst/tests/tests.Rraw:16918]: I tried to trigger the #4402 error in the [`snprintf-windows-test`](https://github.com/Rdatatable/data.table/tree/snprintf-windows-test) branch earlier but wasn't getting anywhere, you can see the 5 commits there used various versions of this, I may have been doing something wrong

**Comment by mattdowle** [comment]
@shrektan Thanks for testing master and I understand it doesn't work. Just so we can get a test going, do you know how to get R on Windows to produce Chinese?  I've tried all kinds of variations of the LANG and LANGUAGE in this PR and none of them seem to generate any Chinese at all.  Is it the wrong string, the wrong environment variable?  Maybe R on Windows must be set to a different language on startup and can't be done at runtime like this?

**Comment by shrektan** [comment]
Thanks for all your efforts. Frankly speaking, I don't really know and I'm not sure if it's possible without the Windows' language being set up to Chinese.

But I will try to find a way (hopefully) this day later when I get back home, where I can set up a Windows machine with English locale to mimic what's happening in the CI.

**Comment by mattdowle** [comment]
https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Localization-of-messages contains (bold mine) : 

> It is usually possible to change the language once R is running via **(not Windows)** Sys.setlocale("LC_MESSAGES", "new_locale"), or by setting an environment variable such as LANGUAGE

**Comment by mattdowle** [comment]
Still no Chinese present in main.Rout. Sigh.
This artile mentioned `chs` and `simplified-chinese` but R, on AppVeyor at least, still remains English.
https://withr.github.io/configure-character-encoding-for-r-under-linux-and-windows/

**Comment by shrektan** [comment]
@mattdowle @MichaelChirico The below code works on my PC (Windows 10, English Language, with Chinese Language installed).
```r
x <- list(
  LC_COLLATE = "Chinese (Simplified)_China.936",
  LC_CTYPE = "Chinese (Simplified)_China.936",
  LC_MONETARY = "Chinese (Simplified)_China.936",
  LC_NUMERIC = "C",
  LC_TIME = "Chinese (Simplified)_China.936"
)
for (i in seq_along(x)) {
  Sys.setlocale(names(x)[[i]], x[[i]])
}
Sys.setenv('LANGUAGE' = 'zh_CN')
```

**Comment by shrektan** [comment]
Looks this approach works. The "main.Rout.fail" prints the expected Chinese message before it gets segfault.

image

**Comment by mattdowle** [comment]
Oh wow! You did it! Not only Chinese but it reproduces the crash too ... awesome!

For future reference, this is what I see in AppVeyor web display; i.e., not Chinese but definitely not ascii, so the translation is kicking in and this is just a web-display issue of the output in AppVeyor that we can ignore for this purpose.
image

**Comment by mattdowle** [comment]
Will merge shortly so that @shrektan can test master to see if it works now please. If it does work now, we can go through and tidy/review in a follow up. If it still doesn't work then we can throw this wrapper away. Although it's a bit hairy, at least it's contained in one file and not too long. And being plain C99, we can be confident it should work on all Windows machines, old and new, MinGW, MSVC, etc.

**Comment by MichaelChirico** [comment]
Regardless of the outcome, thanks for the herculean effort to support this! Awesome stuff.

**Comment by mattdowle** [comment]
@MichaelChirico yes very good point. `%1$` might match  to the 2nd character of `%%1$` which wouldn't be correct. Ok will fix that and add a test.

**Pull request merged**